rustls-wolfcrypt-provider/src/sign/eddsa.rs

```diff
@@ -0,0 +1,169 @@
+
+                // This function reads in an ED25519 private key from the input buffer, input,
+                // parses the private key, and uses it to generate an ed25519_key object,
+                // which it stores in key.
+                ret = unsafe {
+                    wc_Ed25519PrivateKeyDecode(
+                        pkcs8.as_ptr() as *mut u8,
+                        &mut idx,
+                        ed25519_key_object.as_ptr(),
+                        pkcs8_sz,
+                    )
+                };
+                check_if_zero(ret)
+                    .map_err(|_| rustls::Error::General("FFI function failed".into()))?;
+
+                ret = unsafe {
+                    wc_ed25519_make_public(
+                        ed25519_key_object.as_ptr(),
+                        pub_key_raw.as_mut_ptr(),
+                        pub_key_raw_len,
+                    )
+                };
+                check_if_zero(ret)
+                    .map_err(|_| rustls::Error::General("FFI function failed".into()))?;
+
+                ret = unsafe {
+                    wc_ed25519_export_private_only(
+                        ed25519_key_object.as_ptr(),
+                        priv_key_raw.as_mut_ptr(),
+                        &mut priv_key_raw_len,
+                    )
+                };
+                check_if_zero(ret)
+                    .map_err(|_| rustls::Error::General("FFI function failed".into()))?;
+
+                Ok(Self {
+                    priv_key: Arc::new(priv_key_raw.to_vec()),
+                    pub_key: Arc::new(pub_key_raw.to_vec()),
+                    algo: SignatureAlgorithm::ED25519,
+                })
+            }
+            _ => {
+                return Err(rustls::Error::General(
+                    "Unsupported private key format".into(),
+                ))
+            }
+        }
+    }
+}
+
+impl SigningKey for Ed25519PrivateKey {
+    fn choose_scheme(&self, offered: &[SignatureScheme]) -> Option<Box<dyn Signer>> {
+        // Iterate through all ECDSA schemes and check if any is in the offered list
+        ALL_EDDSA_SCHEMES.iter().find_map(|&scheme| {
+            if offered.contains(&scheme) {
+                Some(Box::new(Ed25519Signer {
+                    priv_key: self.priv_key.clone(),
+                    pub_key: self.pub_key.clone(),
+                    scheme: scheme,
+                }) as Box<dyn Signer>)
+            } else {
+                None
+            }
+        })
+    }
+
+    fn algorithm(&self) -> SignatureAlgorithm {
+        self.algo
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Ed25519Signer {
+    priv_key: Arc<Vec<u8>>,
+    pub_key: Arc<Vec<u8>>,
+    scheme: SignatureScheme,
+}
+
+impl Signer for Ed25519Signer {
+    fn sign(&self, message: &[u8]) -> Result<Vec<u8>, rustls::Error> {
+        let mut ret;
+        let message_length: word32 = message.len() as word32;
+        let mut sig: [u8; 1024] = [0; 1024];
```

Comment:
avoid magic numbers, why 1024? There is a ED25519_SIG_SIZE macro for the size needed.